File: web/packages/core/src/providers/__tests__/relay-auth-none.test.ts
import { describe, expect, it } from 'vitest';

import {
  applyDirectCustomHeaders,
  applyDirectQueryParams,
  buildDirectAuthHeaders,
} from '../relay-adapter';

describe('Relay auth=none production request builder', () => {
  const config = {
    apiKey: 'must-not-leak',
    transport: 'openai_chat_completions' as const,
    authMode: 'none' as const,
    headers: [
      { key: 'Authorization', value: 'Bearer must-not-leak' },
      { key: 'X-Private-Key', value: 'must-not-leak' },
    ],
    queryParams: [{ key: 'key', value: 'must-not-leak' }],
  };

  it('injects no authorization or API-key header', () => {
    expect(buildDirectAuthHeaders(config.apiKey, config.authMode)).toEqual({});
    expect(applyDirectCustomHeaders({}, config, () => 'uuid')).toEqual({});
  });

  it('injects no key or custom query parameter', () => {
    expect(applyDirectQueryParams('http://192.168.1.20:8080/v1/models', config))
      .toBe('http://192.168.1.20:8080/v1/models');
  });
});

File: android/app/src/test/java/ai/oriveo/community/core/provider/relay/RelayAuthNoneTest.kt
package ai.oriveo.community.core.provider.relay

import ai.oriveo.community.core.model.ChatRequestOptions
import ai.oriveo.community.core.model.RelayAuthMode
import ai.oriveo.community.core.model.RelayKeyValue
import ai.oriveo.community.core.model.RelayRequestedConfig
import ai.oriveo.community.core.model.RelayTransport
import io.ktor.client.request.HttpRequestBuilder
import org.junit.Assert.assertFalse
import org.junit.Test

class RelayAuthNoneTest {
    @Test
    fun `auth none injects no credential or custom header`() {
        val builder = HttpRequestBuilder()
        with(RelayHeaderBuilder) {
            builder.applyRelayHeaders(
                apiKey = "must-not-leak",
                requestOptions = ChatRequestOptions(
                    relayRequested = RelayRequestedConfig(
                        authMode = RelayAuthMode.None,
                        headers = listOf(
                            RelayKeyValue("Authorization", "Bearer must-not-leak"),
                            RelayKeyValue("X-Private-Key", "must-not-leak"),
                        ),
                        queryParams = listOf(RelayKeyValue("key", "must-not-leak")),
                    ),
                ),
                transport = RelayTransport.OpenAIChatCompletions,
            )
        }

        val names = builder.headers.names().map(String::lowercase).toSet()
        assertFalse("authorization" in names)
        assertFalse("x-api-key" in names)
        assertFalse("x-goog-api-key" in names)
        assertFalse("x-private-key" in names)
    }
}
